plugins/modules/vmware_deploy_ovf.py

```diff
@@ -321,12 +341,31 @@ def __init__(self, module):
         self.entity = None
 
     def get_objects(self):
+        # Get datacenter firstly
         self.datacenter = self.find_datacenter_by_name(self.params['datacenter'])
-        if not self.datacenter:
-            self.module.fail_json(msg='%(datacenter)s could not be located' % self.params)
+        if self.datacenter is None:
+            self.module.fail_json(msg="Datacenter '%(datacenter)s' could not be located" % self.params)
+
+        # Get cluster in datacenter if cluster configured
+        if self.params['cluster']:
+            cluster = self.find_cluster_by_name(self.params['cluster'], datacenter_name=self.datacenter)
+            if cluster is None:
+                self.module.fail_json(msg="Unable to find cluster '%(cluster)s'" % self.params)
+            self.resource_pool = self.find_resource_pool_by_cluster(self.params['resource_pool'], cluster=cluster)
+        # Or get ESXi host in datacenter if ESXi host configured
+        elif self.params['esxi_hostname']:
+            host = self.find_hostsystem_by_name(self.params['esxi_hostname'])
+            if host is None:
+                self.module.fail_json(msg="Unable to find host '%(esxi_hostname)s'" % self.params)
+            self.resource_pool = self.find_resource_pool_by_name(self.params['resource_pool'], folder=host.parent)
```

Comment:
```suggestion
            self.resource_pool = self.find_resource_pool_by_name(self.params['resource_pool'], folder=host.parent)
```

In this case, we can’t deploy correctly an OVF to an ESXi host in a cluster.
Because host.parent is a cluster resource object if an ESXi in a cluster.
So the OVA is deployed randomly on an ESXi host of a cluster.

What do you think?

Reply:
Yeah, before and after this change deploy a template on the specified ESXi host in a cluster is not supported. Before it only support deploy in a cluster or one stand alone ESXi host scenarios, now if there are several stand alone ESXi hosts in datacenter but not in a cluster, you can specify the "esxi_hostname" to deploy the VM. I don't know if there is requirement to support specified ESXi host in a cluster, if there is, we can also add it later.

Reply:
Thank you for responding.

> I don't know if there is requirement to support specified ESXi host in a cluster, if there is, we can also add it later.

Okay.  
I don't think it's a problem.  
Let's consider implementing the feature if requesting :)